### DISCUSSION TITLE
How can I override initial value of a generated provider class?

### DISCUSSION THREAD

**Discussion by rsegecin** [rrousselGit/riverpod#3573] [Q&A]
I'd like to convert the provider below to the new generated class, but the new generated provider doesn't include overrideWithValue that would be preferable as it initializes the provider with the value that you wish without triggering notifiers.

old provider:
```
final appOriginStateProvider = Provider((ref, {eAppOrigin? appOrigin}) {
  return appOrigin ?? eAppOrigin.Player;
});
```

new way to generate:
```
@riverpod
class AppOriginState extends _$AppOriginState {
  @override
  eAppOrigin build({eAppOrigin? appOrigin}) {
  return appOrigin ?? eAppOrigin.Player;
  }
}
```

The AppOriginProvider stays in a library that I made for my App and its accessed by other services also from the same library, so I can't initialize in my App with eg. `ref.watch(appOriginStateProvider(appOrigin: eAppOrigin,User))` because the services from the library would be accessing this provider through `ref.watch(appOriginStateProvider())` giving me another value state. 

To try to initialize the AppOriginProvider I did what's shown below but I couldn't override the provider's initial value.

```
appOriginStateProvider
        .overrideWith(() => AppOriginState(appOrigin: eAppOrigin.Admin));
```

```
@riverpod
class AppOriginState extends _$AppOriginState {
  // ignore: avoid_public_notifier_properties
  eAppOrigin? initialAppOrigin;

  AppOriginState({eAppOrigin? appOrigin}) {
    initialAppOrigin = appOrigin;
  }

  @override
  eAppOrigin build() {
    return  initialAppOrigin ?? eAppOrigin.User;
  }
}

```

How can I override initial value of a generated provider class?

**Comment by normidar** [comment]
You should change the `state` in your `AppOriginState` class, see the docs: https://riverpod.dev/docs/essentials/side_effects#updating-our-local-cache-to-match-the-api-response.

**Reply by rsegecin** [reply]
Thank you for your reply @normidar. Can't do it, this provider is initialized as soon the main MaterialApp starts like so:

    runApp(
      ProviderScope(
        overrides: [
          appOriginStateProvider.overrideWithValue(eAppOrigin.Player),
        ],
        child: const MainApp(),
      ),
    );
    
If I were to listen with `ref.watch(appOriginStateProvider())` and change its state after it would throw an error as it would be trying to reconstruct the widget tree while being built. I could wait for it to be built and then change the state but its not efficient having to build the widget tree twice if you can build it only once.
